client/me/purchases/payment-methods/components/payment-method-edit-dialog.tsx

```diff
@@ -0,0 +1,57 @@
+import { Button, Dialog } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import CardHeading from 'calypso/components/card-heading';
+import Notice from 'calypso/components/notice';
+import type { TranslateResult } from 'i18n-calypso';
+import type { FunctionComponent } from 'react';
+
+import 'calypso/me/purchases/payment-methods/style.scss';
+
+interface Props {
+	paymentMethodSummary: TranslateResult;
+	isVisible: boolean;
+	onClose: () => void;
+	onConfirm: () => void;
+	form: JSX.Element;
+	error: string;
+}
+
+const PaymentMethodEditDialog: FunctionComponent< Props > = ( {
+	paymentMethodSummary,
+	isVisible,
+	onClose,
+	onConfirm,
+	form,
+	error,
+} ) => {
+	const translate = useTranslate();
+
+	return (
+		<Dialog
+			isVisible={ isVisible }
+			additionalClassNames="payment-method-edit-dialog"
+			onClose={ onClose }
+			buttons={ [
+				<Button onClick={ onClose }>{ translate( 'Cancel' ) }</Button>,
+				<Button onClick={ onConfirm } primary>
+					{ translate( 'Save' ) }
+				</Button>,
+			] }
+		>
+			<CardHeading tagName="h2" size={ 24 }>
+				{ translate( 'Update Your Payment Method' ) }
+			</CardHeading>
+			<p>
+				{ translate( 'Please update the following information for {{paymentMethodSummary/}}', {
+					components: {
+						paymentMethodSummary: <strong>{ paymentMethodSummary }</strong>,
+					},
+				} ) }
+			</p>
+			{ error && <Notice status="is-error" isCompact={ true } duration={ 1000 } text={ error } /> }
```

Comment:
The duration doesn't actually do anything, I don't think, if this is being used outside of the notices area, so we can remove it.

```suggestion
			{ error && <Notice status="is-error" isCompact={ true } text={ error } /> }
```

Reply:
I've removed that.